pxf/pxf-jdbc/src/main/java/org/apache/hawq/pxf/plugins/jdbc/JdbcPartitionFragmenter.java

```diff
@@ -0,0 +1,284 @@
+                            break;
+                        case MONTH:
+                            frag_end.add(Calendar.MONTH, curr_interval);
+                            break;
+                        //case YEAR:
+                        default:
+                            frag_end.add(Calendar.YEAR, curr_interval);
+                            break;
+                    }
+                    if (frag_end.after(c_end)) frag_end = (Calendar) c_end.clone();
+
+                    //make metadata of this fragment , converts the date to a millisecond,then get bytes.
+                    byte[] ms_start = ByteUtil.getBytes(frag_start.getTimeInMillis());
+                    byte[] ms_end = ByteUtil.getBytes(frag_end.getTimeInMillis());
+                    byte[] fragmentMetadata = ByteUtil.mergeBytes(ms_start, ms_end);
+
+                    byte[] userData = new byte[0];
+                    Fragment fragment = new Fragment(inputData.getDataSource(), null, fragmentMetadata, userData);
+                    fragments.add(fragment);
+
+                    //continue next fragment.
+                    frag_start = frag_end;
+                }
+                break;
+            }
+            case INT: {
+                int i_start = Integer.parseInt(range[0]);
+                int i_end = Integer.parseInt(range[1]);
+                int curr_interval = intervalNum;
+
+                //validate : curr_interval > 0
+                int frag_start = i_start;
+                while (frag_start < i_end) {
+                    int frag_end = frag_start + curr_interval;
+                    if (frag_end > i_end) frag_end = i_end;
+
+                    byte[] b_start = ByteUtil.getBytes(frag_start);
+                    byte[] b_end = ByteUtil.getBytes(frag_end);
+                    byte[] fragmentMetadata = ByteUtil.mergeBytes(b_start, b_end);
+
+                    byte[] userData = new byte[0];
+                    Fragment fragment = new Fragment(inputData.getDataSource(), null, fragmentMetadata, userData);
+                    fragments.add(fragment);
+
+                    //continue next fragment.
+                    frag_start = frag_end;// + 1;
+                }
+                break;
+            }
+            case ENUM:
+                for (String frag : range) {
+                    byte[] fragmentMetadata = frag.getBytes();
+                    Fragment fragment = new Fragment(inputData.getDataSource(), null, fragmentMetadata, new byte[0]);
+                    fragments.add(fragment);
+                }
+                break;
+        }
+
+        return prepareHosts(fragments);
+    }
+
+    /**
+     * For each fragment , assigned a host address.
+     * In Jdbc Plugin, 'replicas' is the host address of the PXF engine that is running, not the database engine.
+     * Since the other PXF host addresses can not be probed, only the host name of the current PXF engine is returned.

```

Comment:
this limitation probably has performance effect, because all of the fragments will be processed by the same PXF instance. In the future we should probably provide a list of available PXF instances, to make sure we have real parallelism.